test/NuGet.Core.FuncTests/NuGet.Signing.CrossFramework.Test/CrossVerifyTestFixture.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Test.Utility;
+using Org.BouncyCastle.Asn1.X509;
+using Test.Utility.Signing;
+
+namespace NuGet.Signing.CrossFramework.Test
+{
+    public class CrossVerifyTestFixture : IDisposable
+    {
+#if IS_DESKTOP
+        private const string DotnetExe = "dotnet.exe";
+        //In net472 code path, the SDK version and TFM could not be detected automatically, so we manually specified according to the sdk version we're testing against.
+        private const string SdkVersion = "5";
```

Comment:
Do we need to keep updating this in the future?

Reply:
That's a great question! 
I think yes, we need to update it in the future.
I'm not sure if there is any better way to do that.

Usually we only have to test against one version of dotnet(we specified the version(s) in this [line](https://github.com/NuGet/NuGet.Client/blob/dev/build/config.props#L37)), but sometimes we need to insert into multiple versions of dotnet, then we will need to test against multiple version of dotnet, so there will be multiple folders under .\cli\sdk.

During patching dotnet with latest NuGet dlls, we copy from .\cli folder to a temporary testing folder. But we only copy one version under .\cli\sdk.
e.g. If we are testing against  .NET Core3.1 and .NET5, there will be 3.1.100 and 5.0.302 folders under .\cli\sdk. If the test project targets multiple frameworks(like what Dotnet.Integration.Tests used to be), the netcoreapp3.1 test assembly will only copy the 3.1.100 folder from .\cli\sdk to the temporary testing folder and patch it, the net5.0 test assembly will only copy the 5.0.302 folder from .\cli\sdk to the temporary testing folder and patch it.
We could get the same TFM folder with the test assembly dynamically at https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs#L410, for Dotnet.Integration.Test.
However, we could not do the same for test project targets net472(this cross framework test project), which needs to patch dotnet.

Reply:
Having some code that we collectively need to remember to update at some point in the future is fragile.  Can we automate this (e.g.:  T4 template)?  Or, until such time, do we have a playbook we can update with an instruction to update this as necessary?